### DISCUSSION TITLE
Cannot import opus

### DISCUSSION THREAD

**Discussion by jadenquinn** [audacity/audacity#4288] [Q&A]
### Bug description

I had this issue before, but it would just crash Audacity as soon as I try to import an opus file, but now it just says it can't recognise it and that I should import it as raw data, but it's not uncompressed... so that didn't work and would probably damage my speakers.

### Steps to reproduce

1. Open Audacity.
2. Try to import opus.
3. Go on Github to complain after google searching for an hour.

### Expected behavior

I expect Audacity to not only recognise an opus file, but know how to work wiht it, and be able to export an AAC version to boot!

### Actual behavior

It... not doing 5% of that.

### Audacity Version

latest stable version (from audacityteam.org/download)

### Operating system

Windows, macOS

### Additional context

Tested on MacBook Pro (Maxed out, 16-inch i9 2019) on SackOS 12.6.3.

**Comment by LWinterberg** [comment]
Trying under Windows with an [opus example file](https://opus-codec.org/static/examples/ehren-paper_lights-96.opus) - 

* Drag'n'Drop of the file works
* File -> Open works
* File -> Import -> Audio works

Can you open the file I linked too? If not, can you send the zip that Help > Diagnostics > Generate Support Data gives you?

**Reply by jadenquinn** [reply]
Yea no doesn't work. It doesn't crash Audacity anymore but it doesn't know what it is.

**Comment by dozzzzer** [comment]
Testing under Macbook M1 Pro also shows no errors. Opening and playing the [opus example file](https://opus-codec.org/static/examples/ehren-paper_lights-96.opus) works just fine.

**Reply by jadenquinn** [reply]
How did you accomplish this? It just says it can't recognise it. If I try to import it raw, it's just... noise. Really LOUD noise. I'm on a MacBook Pro 2019, so Intel.

**Comment by petersampsonaudacity** [comment]
Works fine for me on W10 with 3.2.4

![image](https://user-images.githubusercontent.com/50205984/217578423-d5423cda-2af0-4bdc-a0cf-527fb55fff55.png)

![image](https://user-images.githubusercontent.com/50205984/217578734-4fece199-d123-40f5-ba4d-cb06ab092944.png)

**Comment by jadenquinn** [comment]
From here (https://github.com/audacity/audacity/issues/4287#issuecomment-1420613620), to answer this question, Audacity doesn't crash when I try importing your file, but it's just as useless, as it just says it doesn't know what the frickle it is. I am on MacOS, however. I do not care how it behaves on Windows though.

**Comment by cpbotha** [comment]
In my case, this was happening because I had not yet configured Audacity to find the `avformat*.dll` shared library from my ffmpeg installation.

To fix this, first install ffmpeg if you have not yet done so:

```shell
winget install "FFmpeg (Shared)"
```

Then make sure that Audacity knows where to find the relevant `avformat*.dll` by going to `Edit | Preferences | Libraries`:

![image](https://github.com/audacity/audacity/assets/937871/79a0d933-bab1-4d1d-9e1d-fd40d045a938)

After doing these steps, I could import Opus files.

Here is the audacity documentation on installing and configuring ffmpeg: https://support.audacityteam.org/basics/installing-ffmpeg

**Comment by jadenquinn** [comment]
As far as I've used this, it works perfectly fine now.
